File: src/main/java/com/ocunha/domain/restaurant/filters/RestaurantNameFilterDefinition.java
package com.ocunha.domain.restaurant.filters;

import com.ocunha.domain.restaurant.model.RestaurantSearchParams;
import com.ocunha.domain.restaurant.model.model.Restaurant;
import org.springframework.stereotype.Component;

@Component
public class RestaurantNameFilterDefinition implements FilterDefinition {
    @Override
    public boolean filter(Restaurant restaurant, RestaurantSearchParams searchParams) {
        return contains(restaurant.name(), searchParams.getName());
    }

}

File: src/main/kotlin/com/ocunha/domain/filter/RestaurantNameFilterDefinition.kt
package com.ocunha.domain.filter

import com.ocunha.domain.restaurant.model.Restaurant
import com.ocunha.domain.restaurant.model.RestaurantSearchParams
import org.springframework.stereotype.Component

@Component
class RestaurantNameFilterDefinition : FilterDefinition {
    override fun filter(restaurant: Restaurant?, searchParams: RestaurantSearchParams?): Boolean {
        return contains(restaurant?.name, searchParams?.name);
    }
}